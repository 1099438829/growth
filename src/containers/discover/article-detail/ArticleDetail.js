import React, { Component } from 'react';
import PropTypes from 'prop-types';
import HtmlView, { NET } from '../../../components/HtmlView';
import Api from '../../../utils/api';

class ArticleDetail extends Component {
  static componentName = 'ArticleDetail';

  static propTypes = {
    uri: PropTypes.string.isRequired,
    dialogContent: PropTypes.string,
  };

  static defaultProps = {
    uri: '',
    dialogContent: '',
  };

  render() {
    return (
      <HtmlView
        type={NET}
        url={Api.ARTICLE_DETAIL.concat(this.props.uri)}
        dialogContent={this.props.dialogContent}
      />);
  }
}
export default ArticleDetail;
