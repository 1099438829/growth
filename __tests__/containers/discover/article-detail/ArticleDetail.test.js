import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import ArticleDetail from '../../../../src/containers/discover/article-detail/ArticleDetail';
import TIPS from '../../../../assets/TIPS';

it('renders correclty', () => {
  const articleDetail = renderer.create(
    <ArticleDetail uri={'articles/javascript.txt'} dialogContent={TIPS[0]} />,
  );
  const articleDetailJson = articleDetail.toJSON();
  expect(articleDetailJson).toMatchSnapshot();
});
