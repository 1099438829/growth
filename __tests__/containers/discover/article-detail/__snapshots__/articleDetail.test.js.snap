// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correclty 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "marginTop": 64,
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,.1)",
            "height": 1334,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 2,
              "flexDirection": "row",
              "justifyContent": "center",
              "minWidth": 180,
              "paddingBottom": 20,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 20,
            }
          }
        >
          <RNSpinkit
            color="#03a9f4"
            size={18}
            style={
              Array [
                Object {
                  "height": 18,
                  "width": 18,
                },
                undefined,
              ]
            }
            type="FadingCircleAlt"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999",
                "marginLeft": 10,
                "maxWidth": 210,
                "minWidth": 150,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Tips : 
            </Text>
            不要复制粘贴代码
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <WebView
    injectedJavaScript=""
    onNavigationStateChange={[Function]}
    scalesPageToFit={true}
    source={
      Object {
        "html": "",
      }
    }
    startInLoadingState={true}
    style={
      Object {
        "backgroundColor": "white",
        "height": 1334,
      }
    }
  />
</View>
`;
